### PR TITLE
Delete inaccessible tasks

### LINKED ISSUE #3362: WS removal: remove unneeded tasks from DB
When we detect the user losing access to a workspace, and after we removed synced TEs we want to remove all tasks that are no longer referred to by any time entry.

### PR DESCRIPTION
**Pull request by juanlaube**
## What's this?
This is a new sync state to delete inaccessible, unreferenced tasks. The state is not connected yet to the state machine.

### Relationships

Closes #3362 

## Why do we want this?
👻 🔥 

## How is it done?
Very similar to the state to delete inaccessible, unreferenced tags 😛 
For the tests, I extracted some common methods to a base class.

### Side effects
None.

## Review hints
Should be a simple review.

## :squid: Permissions
Anyone, whenever, but only after @simonrozsival ✅.

### CODE REVIEW

File: Toggl.Foundation.Tests/Sync/States/CleanUp/DeleteNonReferencedInaccessibleTagsStateTests.cs [hunk 1]
```diff
@@ -32,26 +31,26 @@ public DeleteNonReferencedInaccessibleTagsStateTests()
...
+
+            var tag1 = GetTag(1001, accessibleWorkspace);
+            var tag2 = GetTag(1002, accessibleWorkspace);
+            var tag3 = GetTag(1003, accessibleWorkspace);
+            var tag4 = GetTag(2001, inaccessibleWorkspace);
+            var tag5 = GetTag(2002, inaccessibleWorkspace);
+            var tag6 = GetTag(2003, inaccessibleWorkspace);
+            var tag7 = GetTag(2004, inaccessibleWorkspace);
+
+            var te1 = GetTimeEntry(10001, accessibleWorkspace, SyncStatus.SyncNeeded, new[] { tag1 });
```
Comment (simonrozsival) [inline Toggl.Foundation.Tests/Sync/States/CleanUp/DeleteNonReferencedInaccessibleTagsStateTests.cs]: Why did you change the sync status of the time entry?

Comment (juanlaube) [reply Toggl.Foundation.Tests/Sync/States/CleanUp/DeleteNonReferencedInaccessibleTagsStateTests.cs]: 🙈 ©️ 🍝

Comment (juanlaube) [reply Toggl.Foundation.Tests/Sync/States/CleanUp/DeleteNonReferencedInaccessibleTagsStateTests.cs]: Notice that the sync status for the time entries is irrelevant in this case.

Comment (simonrozsival) [reply Toggl.Foundation.Tests/Sync/States/CleanUp/DeleteNonReferencedInaccessibleTagsStateTests.cs]: Yes, I see. IMO that is one more reason to add different combinations though.
